src/cascadia/TerminalApp/lib/LeafPane.cpp

```diff
@@ -0,0 +1,464 @@
+#include "pch.h"
+#include "LeafPane.h"
+#include "Profile.h"
+#include "CascadiaSettings.h"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::UI;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml::Media;
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace winrt::Microsoft::Terminal::TerminalControl;
+using namespace winrt::Microsoft::Terminal::TerminalConnection;
+using namespace winrt::TerminalApp;
+using namespace TerminalApp;
+
+static const int PaneBorderSize = 2;
+static const int CombinedPaneBorderSize = 2 * PaneBorderSize;
+static const float Half = 0.50f;
+winrt::Windows::UI::Xaml::Media::SolidColorBrush LeafPane::s_focusedBorderBrush = { nullptr };
+winrt::Windows::UI::Xaml::Media::SolidColorBrush LeafPane::s_unfocusedBorderBrush = { nullptr };
+
+LeafPane::LeafPane(const GUID& profile, const TermControl& control, const bool lastFocused) :
+    _control{ control },
+    _lastActive{ lastFocused },
+    _profile{ profile }
+{
+    _root.Children().Append(_border);
+    _border.Child(_control);
+
+    _connectionStateChangedToken = _control.ConnectionStateChanged({ this, &LeafPane::_ControlConnectionStateChangedHandler });
+
+    // On the first Pane's creation, lookup resources we'll use to theme the
+    // LeafPane, including the brushed to use for the focused/unfocused border
+    // color.
+    if (s_focusedBorderBrush == nullptr || s_unfocusedBorderBrush == nullptr)
+    {
+        _SetupResources();
+    }
+
+    // Register an event with the control to have it inform us when it gains focus.
+    _gotFocusRevoker = control.GotFocus(winrt::auto_revoke, { this, &LeafPane::_ControlGotFocusHandler });
+
+    // When our border is tapped, make sure to transfer focus to our control.
+    // LOAD-BEARING: This will NOT work if the border's BorderBrush is set to
+    // Colors::Transparent! The border won't get Tapped events, and they'll fall
+    // through to something else.
+    _border.Tapped([this](auto&, auto& e) {

```

Comment:
This likely needs to capture a strong/weak this and then use it.

Reply:
`_border` is xaml element, and xaml elements don't outlive their containing object

Reply:
OK. I trust you there. I'm just applying a visual pattern to my review. You've been digging deeper in this code than me.